### PR TITLE
Bouts index UI

### PR DESCRIPTION
**Pull request by skanderm**
Adds bouts index page for monitoring recent detections coming in for each feed.

image

## Release Notes

- **New Features**
  - Introduced a new `FeedItem` component to display detailed feed information, including online status and detection statistics.
  - Added a new `BoutsPage` component for displaying and sorting feeds related to bouts.
  - Implemented a new GraphQL query for fetching detections with paginated results.
  - Enhanced the existing `feeds` query to support sorting and additional data fields.

- **Improvements**
  - Updated the `useListenerCount` hook for simplified listener count management.

- **Dependencies**
  - Added new dependencies for enhanced data visualization: `react-vega` and `vega-lite`.

- **Refactor**
  - Renamed main application wrapper from `MyApp` to `OrcasiteApp`.

### CODE REVIEW

File: ui/src/components/Bouts/FeedItem.tsx [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+  const detectionChartData = useMemo(
+    () =>
+      recentDetections.map(({ timestamp, category }) =>
+        // Time ago in minutes
+        ({
+          cat: category,
+          minutesAgo: Math.floor(
+            (now.valueOf() - timestamp.valueOf()) / (60 * 1000),
+          ),
+          // minutesAgo: Math.floor(60 * Math.random()),
```
Comment (dthaler) [inline ui/src/components/Bouts/FeedItem.tsx:78]: Remove commented-out code?

File: ui/src/pages/bouts/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  return (
+    <div>
+      <Head>
+        <title>Bouts | Orcasound</title>
+      </Head>
+
+      <main>
+        {/* <h1>Current bouts</h1>
+        <p>Ongoing bouts (ones without an end time)</p> */}
```
Comment (dthaler) [inline ui/src/pages/bouts/index.tsx:63]: Remove commented out part?

Comment (skanderm) [reply ui/src/pages/bouts/index.tsx]: Thanks, removed!

File: ui/src/components/Bouts/FeedItem.tsx (hunk 1 above)
Comment (skanderm) [reply ui/src/components/Bouts/FeedItem.tsx]: Thanks, removed!

File: ui/src/pages/_app.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ const ReactQueryDevtoolsProd = React.lazy(() =>
 // App needs to be customized in order to make MUI work with SSR
 // https://mui.com/material-ui/integrations/nextjs/#pages-router
 // https://github.com/mui/material-ui/blob/master/examples/material-ui-nextjs-pages-router-ts/pages/_app.tsx
-export default function MyApp(props: AppPropsWithLayout) {
+export default function OrcasiteApp(props: AppPropsWithLayout) {
```
Comment (paulcretu) [inline ui/src/pages/_app.tsx:42]: Any particular reason for this?

File: ui/src/pages/bouts/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import {
+  Box,
+  InputLabel,
+  List,
+  ListItem,
+  MenuItem,
+  Select,
+} from "@mui/material";
```
[the remark this answers is not in the document]
Comment (paulcretu) [reply ui/src/pages/bouts/index.tsx:8]: Should already be optimized with [`optimizePackageImports`](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports)

File: ui/src/components/Bouts/FeedItem.tsx [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+
+// eslint-disable-next-line import/no-unused-modules
+export default function FeedItem({
+  feed,
+  onStatUpdate,
+}: {
+  feed: Pick<Feed, "id" | "name" | "slug" | "online">;
+  onStatUpdate?: (feedId: string, stat: string, value: number) => void;
+}) {
+  const categories: Array<DetectionCategory> = ["WHALE", "VESSEL", "OTHER"];
```
[the remark this answers is not in the document]
Comment (paulcretu) [reply ui/src/components/Bouts/FeedItem.tsx:36]: DetectionCategory is a union type not an enum

File: ui/src/pages/_app.tsx (hunk 3 above)
Comment (skanderm) [reply ui/src/pages/_app.tsx:42]: I just noticed it was generic in the component inspector. Do you want it to stay the same?

File: ui/src/components/Bouts/FeedItem.tsx [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+            </Typography>
+            <Typography display="flex" alignItems="center" fontSize={14}>
+              <Person sx={{ mr: 1, color: theme.palette.accent2.main }} />
+              {listenerCount === undefined ? "-" : listenerCount}
+            </Typography>
+          </Box>
+        </Box>
+      </Box>
+
+      {/* Detections charts */}
```
Comment (paulcretu) [inline ui/src/components/Bouts/FeedItem.tsx:229]: I feel like if you have to comment the section, it should be its own component

**Review by paulcretu — APPROVED** [review]
Looks good, though some seed bouts might be nice to have?

File: ui/src/pages/_app.tsx (hunk 3 above)
Comment (paulcretu) [reply ui/src/pages/_app.tsx:42]: No, it's fine as far as I know it doesn't affect anything, was just wondering

**Comment by skanderm** [comment]
> Looks good, though some seed bouts might be nice to have?

I'll add some seed bouts on the next PR with the Create Bouts page

File: ui/src/components/Bouts/FeedItem.tsx (hunk 6 above)
Comment (skanderm) [reply ui/src/components/Bouts/FeedItem.tsx:229]: Let's extract it once the UI stabilizes

**Pull request merged**